use the screen lock instead of the OPLUS privacy password

Smali changes of vendor_oplus_camera f94a761, over baksmali --sequential-labels output.

diff --git b/classes/com/oplus/encryption/aosp/AospLock.smali b/classes/com/oplus/encryption/aosp/AospLock.smali
new file mode 100644
index 0000000000000000000000000000000000000000..6e68172219a4b177b36fb87075a68a4b40f54e6e
--- /dev/null
+++ b/classes/com/oplus/encryption/aosp/AospLock.smali
@@ -0,0 +1,164 @@
+.class public final Lcom/oplus/encryption/aosp/AospLock;
+.super Ljava/lang/Object;
+.source "AospLock.java"
+
+
+# static fields
+.field private static final CHOOSE:Ljava/lang/String; = "com.oplus.settings.privacy.ChooseGenericPrivacy"
+
+.field private static final CONFIRM:Ljava/lang/String; = "oplus.intent.action.settings.PRIVACY_PWD_CONFIRM"
+
+
+# direct methods
+.method private constructor <init>()V
+    .locals 0
+
+    .line 17
+    invoke-direct {p0}, Ljava/lang/Object;-><init>()V
+
+    .line 18
+    return-void
+.end method
+
+.method public static adapt(Landroid/content/Context;Landroid/content/Intent;)Landroid/content/Intent;
+    .locals 2
+
+    .line 35
+    if-eqz p0, :cond_43
+
+    if-nez p1, :cond_5
+
+    goto :goto_43
+
+    .line 38
+    :cond_5
+    const-string v0, "oplus.intent.action.settings.PRIVACY_PWD_CONFIRM"
+
+    invoke-virtual {p1}, Landroid/content/Intent;->getAction()Ljava/lang/String;
+
+    move-result-object v1
+
+    invoke-virtual {v0, v1}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+
+    move-result v0
+
+    if-eqz v0, :cond_24
+
+    .line 39
+    const-class v0, Landroid/app/KeyguardManager;
+
+    invoke-virtual {p0, v0}, Landroid/content/Context;->getSystemService(Ljava/lang/Class;)Ljava/lang/Object;
+
+    move-result-object p0
+
+    check-cast p0, Landroid/app/KeyguardManager;
+
+    .line 40
+    const/4 v0, 0x0
+
+    if-eqz p0, :cond_20
+
+    invoke-virtual {p0, v0, v0}, Landroid/app/KeyguardManager;->createConfirmDeviceCredentialIntent(Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Landroid/content/Intent;
+
+    move-result-object v0
+
+    .line 41
+    :cond_20
+    if-eqz v0, :cond_23
+
+    move-object p1, v0
+
+    :cond_23
+    return-object p1
+
+    .line 43
+    :cond_24
+    invoke-virtual {p1}, Landroid/content/Intent;->getComponent()Landroid/content/ComponentName;
+
+    move-result-object p0
+
+    if-eqz p0, :cond_42
+
+    invoke-virtual {p1}, Landroid/content/Intent;->getComponent()Landroid/content/ComponentName;
+
+    move-result-object p0
+
+    invoke-virtual {p0}, Landroid/content/ComponentName;->getClassName()Ljava/lang/String;
+
+    move-result-object p0
+
+    const-string v0, "com.oplus.settings.privacy.ChooseGenericPrivacy"
+
+    invoke-virtual {v0, p0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+
+    move-result p0
+
+    if-eqz p0, :cond_42
+
+    .line 44
+    new-instance p0, Landroid/content/Intent;
+
+    const-string p1, "android.app.action.SET_NEW_PASSWORD"
+
+    invoke-direct {p0, p1}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
+
+    return-object p0
+
+    .line 46
+    :cond_42
+    return-object p1
+
+    .line 36
+    :cond_43
+    :goto_43
+    return-object p1
+.end method
+
+.method public static hasLock(Landroid/content/Context;)Z
+    .locals 2
+
+    .line 22
+    const/4 v0, 0x0
+
+    if-nez p0, :cond_4
+
+    .line 23
+    return v0
+
+    .line 25
+    :cond_4
+    const-class v1, Landroid/app/KeyguardManager;
+
+    invoke-virtual {p0, v1}, Landroid/content/Context;->getSystemService(Ljava/lang/Class;)Ljava/lang/Object;
+
+    move-result-object p0
+
+    check-cast p0, Landroid/app/KeyguardManager;
+
+    .line 26
+    if-eqz p0, :cond_15
+
+    invoke-virtual {p0}, Landroid/app/KeyguardManager;->isDeviceSecure()Z
+
+    move-result p0
+
+    if-eqz p0, :cond_15
+
+    const/4 v0, 0x1
+
+    :cond_15
+    return v0
+.end method
+
+.method public static isStandIn(Ljava/lang/String;)Z
+    .locals 1
+
+    .line 31
+    const-string v0, "com.oneplus.filemanager"
+
+    invoke-virtual {v0, p0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+
+    move-result p0
+
+    return p0
+.end method
diff --git a/classes/com/oplus/encryption/common/utils/h0.smali b/classes/com/oplus/encryption/common/utils/h0.smali
index 8a8cf9ba2903d351932c028991afec633b0f57a8..faf74404a88b5bf81c22c7ab0aec1c2377c338f1 100644
--- a/classes/com/oplus/encryption/common/utils/h0.smali
+++ b/classes/com/oplus/encryption/common/utils/h0.smali
@@ -62,11 +62,22 @@
 .method public static a(Landroid/content/Context;Ljava/lang/String;)Z
     .locals 2
 
+    invoke-static {p1}, Lcom/oplus/encryption/aosp/AospLock;->isStandIn(Ljava/lang/String;)Z
+
+    move-result v0
+
+    if-eqz v0, :cond_8
+
+    const/4 p0, 0x1
+
+    return p0
+
     .line 1
+    :cond_8
     const/4 v0, 0x0
 
     .line 2
-    :try_start_1
+    :try_start_9
     invoke-virtual {p0}, Landroid/content/Context;->getPackageManager()Landroid/content/pm/PackageManager;
 
     .line 3
@@ -86,7 +97,7 @@
     const/4 p1, 0x1
 
     .line 11
-    if-eqz p0, :cond_16
+    if-eqz p0, :cond_1e
 
     .line 12
     .line 13
@@ -97,32 +108,32 @@
     and-int/2addr v1, p1
 
     .line 16
-    if-eqz v1, :cond_16
+    if-eqz v1, :cond_1e
 
     .line 17
     .line 18
     iget-boolean p0, p0, Landroid/content/pm/ApplicationInfo;->enabled:Z
-    :try_end_13
-    .catch Landroid/content/pm/PackageManager$NameNotFoundException; {:try_start_1 .. :try_end_13} :catch_14
-    .catch Ljava/lang/NullPointerException; {:try_start_1 .. :try_end_13} :catch_14
+    :try_end_1b
+    .catch Landroid/content/pm/PackageManager$NameNotFoundException; {:try_start_9 .. :try_end_1b} :catch_1c
+    .catch Ljava/lang/NullPointerException; {:try_start_9 .. :try_end_1b} :catch_1c
 
     .line 19
     .line 20
     return p0
 
     .line 21
-    :catch_14
+    :catch_1c
     move-exception p0
 
     .line 22
-    goto :goto_17
+    goto :goto_1f
 
     .line 23
-    :cond_16
+    :cond_1e
     return p1
 
     .line 24
-    :goto_17
+    :goto_1f
     const-string p1, "[isAppEnable] e:"
 
     .line 25
@@ -170,7 +181,18 @@
 .method public static b(Landroid/content/Context;Ljava/lang/String;)Z
     .locals 3
 
+    invoke-static {p1}, Lcom/oplus/encryption/aosp/AospLock;->isStandIn(Ljava/lang/String;)Z
+
+    move-result v0
+
+    if-eqz v0, :cond_8
+
+    const/4 p0, 0x1
+
+    return p0
+
     .line 1
+    :cond_8
     invoke-static {p1}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
 
     .line 2
@@ -186,7 +208,7 @@
     const/4 v2, 0x0
 
     .line 8
-    if-eqz v0, :cond_f
+    if-eqz v0, :cond_17
 
     .line 9
     .line 10
@@ -202,8 +224,8 @@
     return v2
 
     .line 16
-    :cond_f
-    :try_start_f
+    :cond_17
+    :try_start_17
     invoke-virtual {p0}, Landroid/content/Context;->getPackageManager()Landroid/content/pm/PackageManager;
 
     .line 17
@@ -218,11 +240,11 @@
     .line 22
     .line 23
     move-result-object p0
-    :try_end_17
-    .catch Ljava/lang/Exception; {:try_start_f .. :try_end_17} :catch_1c
+    :try_end_1f
+    .catch Ljava/lang/Exception; {:try_start_17 .. :try_end_1f} :catch_24
 
     .line 24
-    if-eqz p0, :cond_1b
+    if-eqz p0, :cond_23
 
     .line 25
     .line 26
@@ -232,11 +254,11 @@
     return p0
 
     .line 28
-    :cond_1b
+    :cond_23
     return v2
 
     .line 29
-    :catch_1c
+    :catch_24
     move-exception p0
 
     .line 30
diff --git a/classes/com/oplus/encryption/common/utils/m0.smali b/classes/com/oplus/encryption/common/utils/m0.smali
index 080c3dc6a9219e49c2505300fe543e7703a303ee..18d2bc1fcdea9ee3a6b82cd1d2e58e8f82442585 100644
--- a/classes/com/oplus/encryption/common/utils/m0.smali
+++ b/classes/com/oplus/encryption/common/utils/m0.smali
@@ -50,198 +50,13 @@
 .end method
 
 .method public static final a(Landroid/content/Context;)Z
-    .locals 8
-
-    .line 1
-    const/4 v1, 0x0
-
-    .line 2
-    if-eqz p0, :cond_47
-
-    .line 3
-    .line 4
-    invoke-virtual {p0}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
-
-    .line 5
-    .line 6
-    .line 7
-    move-result-object v2
-
-    .line 8
-    sget-object p0, Lcom/oplus/encryption/common/utils/m0;->a:Landroid/net/Uri;
-
-    .line 9
-    .line 10
-    const-string v0, "state_result"
-
-    .line 11
-    .line 12
-    invoke-static {p0, v0}, Landroid/net/Uri;->withAppendedPath(Landroid/net/Uri;Ljava/lang/String;)Landroid/net/Uri;
-
-    .line 13
-    .line 14
-    .line 15
-    move-result-object v3
-
-    .line 16
-    const/4 v6, 0x0
-
-    .line 17
-    const/4 v7, 0x0
-
-    .line 18
-    const/4 p0, 0x0
-
-    .line 19
-    const/4 v4, 0x0
-
-    .line 20
-    const/4 v5, 0x0
-
-    .line 21
-    :try_start_14
-    invoke-virtual/range {v2 .. v7}, Landroid/content/ContentResolver;->query(Landroid/net/Uri;[Ljava/lang/String;Ljava/lang/String;[Ljava/lang/String;Ljava/lang/String;)Landroid/database/Cursor;
-
-    .line 22
-    .line 23
-    .line 24
-    move-result-object p0
-
-    .line 25
-    if-eqz p0, :cond_2c
-
-    .line 26
-    .line 27
-    invoke-interface {p0}, Landroid/database/Cursor;->moveToNext()Z
-
-    .line 28
-    .line 29
-    .line 30
-    move-result v0
-
-    .line 31
-    if-eqz v0, :cond_2c
-
-    .line 32
-    .line 33
-    invoke-interface {p0, v1}, Landroid/database/Cursor;->getInt(I)I
-
-    .line 34
-    .line 35
-    .line 36
-    move-result v0
-    :try_end_24
-    .catch Ljava/lang/Exception; {:try_start_14 .. :try_end_24} :catch_2a
-    .catchall {:try_start_14 .. :try_end_24} :catchall_28
-
-    .line 37
-    if-lez v0, :cond_2c
-
-    .line 38
-    .line 39
-    const/4 v1, 0x1
-
-    .line 40
-    goto :goto_2c
-
-    .line 41
-    :catchall_28
-    move-exception v0
-
-    .line 42
-    goto :goto_41
-
-    .line 43
-    :catch_2a
-    move-exception v0
-
-    .line 44
-    goto :goto_32
-
-    .line 45
-    :cond_2c
-    :goto_2c
-    if-eqz p0, :cond_31
-
-    .line 46
-    .line 47
-    invoke-interface {p0}, Landroid/database/Cursor;->close()V
-
-    .line 48
-    .line 49
-    .line 50
-    :cond_31
-    return v1
-
-    .line 51
-    :goto_32
-    :try_start_32
-    const-string v2, "SettingsPasswordUtil"
-
-    .line 52
-    .line 53
-    invoke-virtual {v0}, Ljava/lang/Throwable;->getMessage()Ljava/lang/String;
-
-    .line 54
-    .line 55
-    .line 56
-    move-result-object v0
-
-    .line 57
-    invoke-static {v2, v0}, Lp8/b;->c(Ljava/lang/String;Ljava/lang/String;)V
-    :try_end_3b
-    .catchall {:try_start_32 .. :try_end_3b} :catchall_28
-
-    .line 58
-    .line 59
-    .line 60
-    if-eqz p0, :cond_47
-
-    .line 61
-    .line 62
-    invoke-interface {p0}, Landroid/database/Cursor;->close()V
-
-    .line 63
-    .line 64
-    .line 65
-    goto :goto_47
-
-    .line 66
-    :goto_41
-    if-eqz p0, :cond_46
+    .locals 1
 
-    .line 67
-    .line 68
-    invoke-interface {p0}, Landroid/database/Cursor;->close()V
+    invoke-static {p0}, Lcom/oplus/encryption/aosp/AospLock;->hasLock(Landroid/content/Context;)Z
 
-    .line 69
-    .line 70
-    .line 71
-    :cond_46
-    throw v0
+    move-result p0
 
-    .line 72
-    :cond_47
-    :goto_47
-    return v1
-    .line 73
-    .line 74
-    .line 75
-    .line 76
-    .line 77
-    .line 78
-    .line 79
-    .line 80
-    .line 81
-    .line 82
-    .line 83
-    .line 84
-    .line 85
-    .line 86
-    .line 87
-    .line 88
-    .line 89
-    .line 90
+    return p0
 .end method
 
 .method public static final b(Lcom/oplus/encryption/main/activity/FileSafeActivity;Lcom/oplus/encryption/main/activity/FileSafeActivity$a;)Lkotlinx/coroutines/c1;
diff --git a/classes/com/oplus/encryption/main/activity/FileSafeActivity.smali b/classes/com/oplus/encryption/main/activity/FileSafeActivity.smali
index 58c30227066689d385870676b9b5068da996c661..2fee5945b4415f71886aefd2799cdf28ec1d6e7c 100644
--- a/classes/com/oplus/encryption/main/activity/FileSafeActivity.smali
+++ b/classes/com/oplus/encryption/main/activity/FileSafeActivity.smali
@@ -1501,6 +1501,10 @@
     const/4 v1, 0x4
 
     .line 21
+    invoke-static {p0, v0}, Lcom/oplus/encryption/aosp/AospLock;->adapt(Landroid/content/Context;Landroid/content/Intent;)Landroid/content/Intent;
+
+    move-result-object v0
+
     invoke-virtual {p0, v0, v1}, Landroidx/activity/ComponentActivity;->startActivityForResult(Landroid/content/Intent;I)V
 
     .line 22
@@ -1514,7 +1518,7 @@
 
     .line 27
     .line 28
-    if-eqz v0, :cond_21
+    if-eqz v0, :cond_25
 
     .line 29
     .line 30
@@ -1526,7 +1530,7 @@
     .line 32
     .line 33
     .line 34
-    :cond_21
+    :cond_25
     return-void
     .line 35
     .line 36
@@ -1637,6 +1641,10 @@
     const/4 v1, 0x3
 
     .line 21
+    invoke-static {p0, v0}, Lcom/oplus/encryption/aosp/AospLock;->adapt(Landroid/content/Context;Landroid/content/Intent;)Landroid/content/Intent;
+
+    move-result-object v0
+
     invoke-virtual {p0, v0, v1}, Landroidx/activity/ComponentActivity;->startActivityForResult(Landroid/content/Intent;I)V
 
     .line 22
@@ -2029,7 +2037,7 @@
     const/4 v4, 0x1
 
     .line 30
-    if-eqz p1, :cond_7e
+    if-eqz p1, :cond_82
 
     .line 31
     .line 32
@@ -2041,7 +2049,7 @@
     move-result p1
 
     .line 36
-    if-nez p1, :cond_b0
+    if-nez p1, :cond_b8
 
     .line 37
     .line 38
@@ -2158,17 +2166,21 @@
     .line 90
     .line 91
     :try_start_5a
+    invoke-static {p0, v7}, Lcom/oplus/encryption/aosp/AospLock;->adapt(Landroid/content/Context;Landroid/content/Intent;)Landroid/content/Intent;
+
+    move-result-object v7
+
     invoke-virtual {p0, v7, v4}, Landroidx/activity/ComponentActivity;->startActivityForResult(Landroid/content/Intent;I)V
-    :try_end_5d
-    .catch Ljava/lang/Exception; {:try_start_5a .. :try_end_5d} :catch_5e
+    :try_end_61
+    .catch Ljava/lang/Exception; {:try_start_5a .. :try_end_61} :catch_62
 
     .line 92
     .line 93
     .line 94
-    goto :goto_68
+    goto :goto_6c
 
     .line 95
-    :catch_5e
+    :catch_62
     move-exception p1
 
     .line 96
@@ -2189,7 +2201,7 @@
 
     .line 104
     .line 105
-    :goto_68
+    :goto_6c
     sget-object p1, Lcom/oplus/encryption/common/utils/d;->a:Ljava/lang/Object;
 
     .line 106
@@ -2206,7 +2218,7 @@
 
     .line 112
     .line 113
-    if-eqz p1, :cond_76
+    if-eqz p1, :cond_7a
 
     .line 114
     .line 115
@@ -2218,7 +2230,7 @@
 
     .line 118
     .line 119
-    :cond_76
+    :cond_7a
     sget p1, Ljd/a;->coui_zoom_fade_enter:I
 
     .line 120
@@ -2232,10 +2244,10 @@
     .line 124
     .line 125
     .line 126
-    goto :goto_b0
+    goto :goto_b8
 
     .line 127
-    :cond_7e
+    :cond_82
     invoke-direct {p0}, Lcom/oplus/encryption/main/activity/FileSafeActivity;->closeCloudSyncSwitch()V
 
     .line 128
@@ -2275,10 +2287,14 @@
     const/4 v0, 0x2
 
     .line 148
-    :try_start_93
+    :try_start_97
+    invoke-static {p0, p1}, Lcom/oplus/encryption/aosp/AospLock;->adapt(Landroid/content/Context;Landroid/content/Intent;)Landroid/content/Intent;
+
+    move-result-object p1
+
     invoke-virtual {p0, p1, v0}, Landroidx/activity/ComponentActivity;->startActivityForResult(Landroid/content/Intent;I)V
-    :try_end_96
-    .catch Ljava/lang/Exception; {:try_start_93 .. :try_end_96} :catch_98
+    :try_end_9e
+    .catch Ljava/lang/Exception; {:try_start_97 .. :try_end_9e} :catch_a0
 
     .line 149
     .line 150
@@ -2286,10 +2302,10 @@
     move v3, v4
 
     .line 152
-    goto :goto_a0
+    goto :goto_a8
 
     .line 153
-    :catch_98
+    :catch_a0
     move-exception p1
 
     .line 154
@@ -2306,7 +2322,7 @@
     .line 159
     .line 160
     .line 161
-    :goto_a0
+    :goto_a8
     sput-boolean v3, Lcom/oplus/encryption/common/utils/d;->c:Z
 
     .line 162
@@ -2327,7 +2343,7 @@
     .line 169
     .line 170
     .line 171
-    if-eqz v3, :cond_b0
+    if-eqz v3, :cond_b8
 
     .line 172
     .line 173
@@ -2339,8 +2355,8 @@
 
     .line 176
     .line 177
-    :cond_b0
-    :goto_b0
+    :cond_b8
+    :goto_b8
     return-void
     .line 178
     .line 179
diff --git a/classes/com/oplus/encryption/settings/EncryptionSettingsFragment.smali b/classes/com/oplus/encryption/settings/EncryptionSettingsFragment.smali
index 87e0db705c6fcfe3bf11eba2d45f4e564b8f4da1..7c31a99cb552f610650360b2ee9c0df7615a3e8a 100644
--- a/classes/com/oplus/encryption/settings/EncryptionSettingsFragment.smali
+++ b/classes/com/oplus/encryption/settings/EncryptionSettingsFragment.smali
@@ -3349,7 +3349,7 @@
 
     .line 259
     .line 260
-    const-string v0, "oplus.intent.action.settings.PRIVACY_PWD_SETTINGS"
+    const-string v0, "android.settings.SECURITY_SETTINGS"
 
     .line 261
     .line 262
